### ISSUE TITLE
Config flows does not send empty fields

### ISSUE THREAD

**Issue by bdraco** [home-assistant/frontend#9506]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Empty fields are not sent in the config flow which prevents clearing values

### Describe the behavior you expected

Empty fields are sent as ''

### Steps to reproduce the issue

When submitting a config flow, if a field is empty, it is omitted from the post.

1. Edit nmap options
<img width="443" alt="Screen Shot 2021-07-05 at 11 28 34 AM" src="https://user-images.githubusercontent.com/663432/124499776-33d5c000-dd84-11eb-9d88-fd83433a7515.png">

2. Next clear the exclude field
<img width="450" alt="Screen Shot 2021-07-05 at 11 28 39 AM" src="https://user-images.githubusercontent.com/663432/124499795-3df7be80-dd84-11eb-91f0-0516689ec189.png">

3. Notice the payload is missing `exclude`
<img width="563" alt="Screen Shot 2021-07-05 at 11 28 53 AM" src="https://user-images.githubusercontent.com/663432/124499814-46e89000-dd84-11eb-9490-f718367d2c40.png">

### What version of Home Assistant Core has the issue?

dev

**Comment by bdraco** [comment]
#1662 explicitly excludes them

**Comment by bdraco** [comment]
This came up again while working on unifiprotect

**Comment by AngellusMortis** [comment]
Yeah, came up during testing while adding a new option for UniFi Protect: https://github.com/home-assistant/core/pull/77375. The `ignored_devices` option is a CSV of MAC addresses. If you empty out the field, it passes the original value back to the backend and never removes the address. You have to set it to `" "` to clear the field.

**Comment by frenck** [comment]
<https://github.com/home-assistant/core/pull/77375#pullrequestreview-1087372004>

**Comment by AngellusMortis** [comment]
What is not really said in that comment is that this is specifically an issue with using `default=` for options instead of `description={"suggested_value":`. 

Which it looks like far more integrations are using `default` to prepopulate instead of suggested value:

![image](https://user-images.githubusercontent.com/490848/187034108-1d0483d0-1369-4fe7-85dc-01443d8d1b18.png)

![image](https://user-images.githubusercontent.com/490848/187034159-cd6bbbbe-9d7d-458a-af1e-87b35860358e.png)
